test_util/test_runner.py

```diff
@@ -64,55 +56,29 @@ def integration_test(
         for key, value in add_env.items():
             extra_env = key + '=' + value
             test_env.append(extra_env)
-    test_env = ['export '+e+'\n' for e in test_env]
-    test_env = ''.join(test_env)
-    test_cmd = 'py.test -vv ' + ci_flags + ' /integration_test.py'
-
-    log.info('Building py.test image')
-    # Make a clean docker context
-    temp_dir = tempfile.mkdtemp()
-    cmd_script = """
-#!/bin/bash
-set -euo pipefail; set -x
-{test_env}
-{test_cmd}
-""".format(test_env=test_env, test_cmd=test_cmd)
-    write_string(join(temp_dir, 'test_wrapper.sh'), cmd_script)
-    check_call(['cp', test_script, join(temp_dir, 'integration_test.py')])
-    check_call(['cp', pytest_docker, join(temp_dir, 'Dockerfile')])
-    check_call(['cp', test_server, join(temp_dir, 'test_server.py')])
-    check_call(['cp', test_server_docker, join(temp_dir, 'test_server_Dockerfile')])
-    check_call(['docker', 'build', '-t', 'py.test', temp_dir])
 
-    log.info('Exporting py.test image')
-    pytest_image_tar = 'DCOS_integration_test.tar'
-    check_call(['docker', 'save', '-o', join(temp_dir, pytest_image_tar), 'py.test'])
-
-    log.info('Transferring py.test image')
-    tunnel.write_to_remote(join(temp_dir, pytest_image_tar), join(test_dir, pytest_image_tar))
-    log.info('Loading py.test image on remote host')
-    tunnel.remote_cmd(['docker', 'load', '-i', join(test_dir, pytest_image_tar)])
+    test_wrapper = """#!/bin/bash
+{env}
+cd /opt/mesosphere/active/dcos-integration-test
+py.test -vv {ci_flags}
+"""
+    test_env_str = ''.join(['export '+e+'\n' for e in test_env])
+    write_string('test_wrapper.sh', test_wrapper.format(env=test_env_str, ci_flags=ci_flags))
 
-    test_container_name = 'int_test_' + str(int(time.time()))
-    docker_cmd = ['docker', 'run', '--net=host', '--name='+test_container_name, 'py.test']
     try:
+        wrapper_path = join(test_dir, 'test_wrapper.sh')
         log.info('Running integration test...')
-        tunnel.remote_cmd(docker_cmd, timeout=timeout)
+        tunnel.write_to_remote('test_wrapper.sh', wrapper_path)
+        dump_logs(tunnel.remote_cmd(['bash', wrapper_path], timeout=timeout))
         log.info('Successful test run!')
-    except CalledProcessError:
+    except CalledProcessError as e:
         log.exception('Test failed!')
+        dump_logs(e.output)
```

Comment:
The original version logged this directly back, and https://github.com/dcos/dcos/pull/325/files/7938eb21e5756b9ca458e0156ed1cce4c0457651#diff-35acaec42b244e7ea0b6a2474e06aa05R94 also does that, and things work.

Not streaming out logs makes it really impossible to debug things on the integration test run. It shouldn't have gotten changed to not real time responding / that is something that was definitely missed in a previous review and needs to be fixed.

The teamcity messages themselves don't use any special ascii characters or escape codes: https://github.com/JetBrains/teamcity-messages/blob/master/teamcity/messages.py#L42.

Just outputs a ascii string which has the information in a format TeamCity understands.